tests/DCPS/Ownership/DataReaderListener.cpp

```diff
@@ -101,22 +101,25 @@ void DataReaderListenerImpl::on_requested_deadline_missed(
   DDS::DataReader_ptr,
   const DDS::RequestedDeadlineMissedStatus & status)
 {
+  ACE_Guard<ACE_Thread_Mutex> guard(mutex_);
   ACE_DEBUG((LM_DEBUG, ACE_TEXT("(%P|%t) %s: on_requested_deadline_missed(): ")
           ACE_TEXT(" handle %d total_count_change %d\n"),
-          this->reader_id_, status.last_instance_handle, status.total_count_change));
+          reader_id_, status.last_instance_handle, status.total_count_change));
 }
 
 void DataReaderListenerImpl::on_requested_incompatible_qos(
   DDS::DataReader_ptr,
   const DDS::RequestedIncompatibleQosStatus &)
 {
+  ACE_Guard<ACE_Thread_Mutex> guard(mutex_);
```

Comment:
Do this and the other no-op callbacks (on_subscription_matched, etc.) need to lock?  I guess it doesn't hurt

Reply:
I included them in case we add code here eventually, but no technically these don't need locks. at the moment.